job/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -291,13 +299,39 @@ public void setQuietPeriod(Integer seconds) throws IOException {
     }
 
     @Override public CauseOfBlockage getCauseOfBlockage() {
-        // TODO is there any legitimate reason to block a flow? maybe !isConcurrentBuild()?
+        if (isLogUpdated() && !isConcurrentBuild()) {
```

Comment:
Yes, this core's code is very suspicious. I've investigated it several times, but didn't find obvious issues caused by it in JIRA.